Update and migrate translating dic/ini files from wiki (#16627)
nvaccess/nvda: projectDocs/dev/developerGuide/developerGuide.md

This is a UTF-8 encoded text file.
Blank lines and lines beginning with a "#" character are ignored.
All other lines should contain a character, followed by a tab, then one or more descriptions separated by tabs.

> Multiple descriptions for a character will be read with natural pauses between them.

Reviewer: Not exactly. My understanding (with tests) is that reading one character (e.g. numpad2, left/right arrows, pavnum1/3) read all the descriptions separated by a pause. But when many characters are read successively in spelling commands (e.g. triple press of numpad5, numpad8 or NVDA+upArrow), only the first description is used.

@cary-rowen, @hwf1324 could you explain the use case of multiple descriptions in the zh_CN file? Why one is not enough?

Reviewer 2: Yes, one character description is not enough in Chinese, currently our first description is a short description of the character, which serves as a quick way to determine the meaning of the Chinese character, while the second description is a detailed explanation of the exact meaning of the character and examples. Unlike alphabets, Chinese characters are ideographs, and there may be very rich and large amount of information in a single character.

Of course as you described in your comment, this is currently very important when reading multiple kanji in a row, otherwise the spelling would become very lengthy.

Although there are multiple representations of a kanji in the long description, we also seem to be using only two descriptions at the moment (a short description and a long description). For the line you commented on, I don't seem to feel a pause between the two descriptions when reading the description of a kanji.

Sorry for the possible wording bias in my explanation due to the use of translations.

Author: How does this read?

```suggestion
Multiple descriptions for a character will be read with natural pauses between them when reading a single character.
When reading character descriptions of multiple subsequent characters, the first description is used for each character.
```

> 

Author: My understanding (with testing) is each description is read out after each other with pauses in between

For example:

```
# This is a comment.
a	alpha
b	bravo beta
```

In this example, "a" will read "alpha" as the character description, and "b" will read as "bravo, beta".

In most cases, the characters in this file should be a single lower case character.
It is assumed that characters will have the same description regardless of their case, so upper case characters are converted to lower case before looking up their character descriptions.

#### Translating this file {#TranslatingCharacterDescriptionsFile}
Translations for characterDescriptions.dic happen on SVN following [this process](https://github.com/nvaccess/nvda/wiki/TranslatingUsingAutomaticProcess).

> For a full example and reference, please look at the English [characterDescriptions.dic file](https://github.com/nvaccess/nvda/blob/master/source/locale/en/characterDescriptions.dic)

Author: ```suggestion
For a full example and reference, please look at the English [characterDescriptions.dic file](https://github.com/nvaccess/nvda/blob/master/source/locale/en/characterDescriptions.dic).
```

The second section provides information about when and how to pronounce all symbols.
It begins with the line:

    symbols:
```
symbols:
```

Subsequent lines should contain several fields separated by tabs.
The only mandatory fields are the identifier and replacement.
The default will be used for omitted fields.
The fields are as follows:

* identifier: The identifier of the symbol.
In most cases, this is just the character or characters of the symbol.
However, it can also be the identifier of a complex symbol.
Certain characters cannot be typed into the file, so the following special sequences can be used:
  * \0: null
  * \t: tab
  * \n: line feed
  * \r: carriage return
  * \f: form feed
  * \#: # character (needed because # at the start of a line denotes a comment)
* replacement: The text which should be spoken for the symbol.

> If the symbol is a complex symbol, \1, \2, etc. can be used to refer to the groups matches, which will be inlined in the replacement, allowing for simpler rules.

Reviewer: Why have you removed this line and the following one? This information is still useful to have.

Author: It wasn't in the wiki, and was lost when reformatting. I'll restore this

If the gesture to be remapped is a global command, you may execute the following steps to find out the class and the script name of the command:

* activate input help (`NVDA+1`)
* press the gesture you want to remap, e.g. `NVDA+]` (laptop layout)
* de-activate input help (`NVDA+1`)
* open the log (`NVDA+F1`)
* find out the line corresponding to the moment you have executed the gesture, e.g.:
  ```
  Input help: gesture kb(laptop):NVDA+], bound to script rightMouseClick on globalCommands.GlobalCommands
  ```

The information you are searching is on this line:

* script name: `rightMouseClick`
* class name: `globalCommands.GlobalCommands` (Note that this is always this class for global commands)
* original gesture: `kb(laptop):NVDA+]`

##### Case of an application specific script {#TranslatingGesturesStepCaseApplication}

In case you want to remap an application specific script, you will have to follow the same steps as those for a global command script.
You just need to ensure before proceeding that you are in the targeted application.

##### Case of an object specific script {#TranslatingGesturesStepCaseObject}

> Object specific scripts such as the ones linked to `NVDAObjects.window.winword.WordDocument` do not have generally a help message.
> So the previous technique will not work.
> You will need to explore NVDA's source code to find the class and the script name of the gesture you want to remap.

Reviewer: It's not true that the previous technique will not work. Thus I suggest the following replacement:

```suggestion
For object specific scripts such as the ones linked to `NVDAObjects.window.winword.WordDocument`, you may follow the same steps as those for application specific script, paying attention to the two following points:
* You need to ensure before proceeding that the object to which the script is bound is focused.
* Some of these script have no help message, so you may not hear anything when executing them in input help mode; but the script's name and the class of the object will still appear in the log.

Though, the class  of the object appearing in the log may be a subclass of the one where the original gesture is actually bound. In this case, you will have to explore NVDA's source code to find this parent class.
```
